File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.java
class Solution {
    public String solution(String s) {
        String answer = "";
        int idx = 0;

        for(char c : s.toCharArray()){
            if(c == ' '){
                answer += " ";
                idx = 0;
            } else {
                answer += idx % 2 == 0 ? Character.toUpperCase(c) : Character.toLowerCase(c);
                idx++;
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.js
function solution(s) {
    let answer = "";
    let idx = 0;
    
    for(const c of s){
        if(c == ' '){
            answer += " ";
            idx = 0;
        } else {
            answer += idx % 2 == 0 ? c.toUpperCase() : c.toLowerCase();
            idx++;
        }
    }
    return answer;
}